### DISCUSSION TITLE
ComboBox inside a DataGrid Doesn't Always Capture Pointer Events

### DISCUSSION THREAD

**Discussion by tombo86** [AvaloniaUI/Avalonia#8286] [Q&A]
I have a DataGrid that has a column defined as:

```
<DataGridTemplateColumn	CellTemplate="{StaticResource LongPressColumnTemplate}"
                                                CellEditingTemplate="{StaticResource LongPressColumnTemplate}"
						Header="Enabled"
						Width="70" />
```

The ColumnTemplate is defined as:

```
<DataTemplate x:Key="LongPressModeColumnTemplate" DataType="viewModels:PedalConfigViewModel">
	<ComboBox Items="{Binding LongPressModes, Mode=OneTime}"
                              Width="80"
                              SelectedItem="{Binding LongPressMode}">
	</ComboBox>
</DataTemplate>
```

When I run the application, the ComboBox requires two clicks to open the drop down.  Usually, if I select an item in the drop down, the dropdown closes, but the item is not selected.  If I reopen the drop down, it only takes one click, and selecting an item works.  

Looking at the sequence of events in Avalonia DevTools, it appears that the DataGrid is capturing the pointer tunneling events before the ComboBox even has a chance to respond to them.

I am on Windows 11, using Visual Studio 2022.

**Comment by maxkatz6** [comment]
Hi! I can't repro with your template. Also, I don't see any tunneling events captures in DataGrid.

https://user-images.githubusercontent.com/3163374/172965192-e2603a8c-64b2-4f0a-b68e-af52e485e448.mp4

![image](https://user-images.githubusercontent.com/3163374/172965454-0f6c91c4-2db7-4b7d-9357-2081d1c8dc2f.png)

**Comment by maxkatz6** [comment]
Also, you don't need to have both CellEditingTemplate and CellTemplate if template is the same, keep just CellTemplate to avoid double template creation.

**Comment by tombo86** [comment]
Thanks for checking it out, maxkatz6.  I am using the Fluent theme.  Are you using a different one?  Maybe that is the issue.

**Reply by maxkatz6** [reply]
No, fluent

**Reply by maxkatz6** [reply]
But I checked on master branch. It could have been already fixed.

**Comment by tombo86** [comment]
Interesting.  Maybe I will check that out.  One way or another, glad to hear it is working for you.  That gives hope that I can fix it as I have been banging my head against the wall with this for days.
